_posts/en/newsletters/2019-09-18-newsletter.md

```diff
@@ -0,0 +1,312 @@
+---
+title: 'Bitcoin Optech Newsletter #64'
+permalink: /en/newsletters/2019/09/18/
+name: 2019-09-18-newsletter
+slug: 2019-09-18-newsletter
+type: newsletter
+layout: newsletter
+lang: en
+---
+This week's newsletter describes an idea for providing SPV fraud proofs
+without changing Bitcoin's consensus rules and summarizes several talks
+from the Bitcoin Edge Dev++ training sessions and Scaling Bitcoin
+conference held last week in Tel Aviv.  Also included is our regular
+section on notable changes to popular Bitcoin infrastructure projects.
+
+{% comment %}<!-- include references.md below the fold but above any Jekyll/Liquid variables-->{% endcomment %}
+{% include references.md %}
+
+## Action items
+
+*None this week.*
+
+## News
+
+- **SPV fraud proofs without a soft fork:** Ruben Somsen started a
+  [discussion][fraud proofs] on the Bitcoin-Dev mailing list about fraud
+  proofs based on the [utreexo][] UTXO accumulator.  This would allow SPV
+  clients with at least one honest connection to determine whether or
+  not any blocks on different sides of a block chain fork contained
+  invalid transactions.
+
+    The technique works by first detecting whether different nodes are
+    reporting different blocks at the same block height, indicating a
+    fork.  Then the client requests each of its peers commit to the UTXO
+    state from the block before the fork via a utreexo merkle root.
+    After that commitment, the client downloads the fork block, copies
+    of the UTXOs it spends, and the partial merkle branches connecting
+    the utreexo merkle root to the spent UTXOs.  This allows the SPV
+    client to verify that each fork block only spends UTXOs that existed
+    at the time the block was created.  The client can then perform the
+    other consensus checks on block validity which don't depend on state
+    (e.g., all the fields in its block header are correct, none of its
+    outputs spends more than 21 million BTC, etc...).  With all that
+    information, the SPV client can determine whether any fork blocks
+    are invalid (and, if so, it can reject them and any of their
+    descendants).
+
+    In case any peers disagree about the value of the utreexo state at
+    the pre-fork block, the SPV client can work with them to find a
+    transition from a state they both agreed upon to a state they find
+    disagreeable and can perform the same procedure.  In the worst case,
+    this increases the cost of the fraud proof by a few megabytes per
+    lying peer.
```

Comment:
"this increases the cost of the fraud proof by a few megabytes per lying peer." How did you get to this figure? I don't see it in the mailing list post.

Reply:
The mailing list post has estimates for the cost of verifying the state transition from the pre-fork block to one of the fork blocks:

```text
1. block N+1 from the most PoW chain (~1-2MB)
2. the utreexo hash commitment inside of block N (e.g. a merkle path
to the coinbase)
3. the utreexo merkle proofs which prove that all inputs of N+1 are
part of the UTXO set (~1MB)
```

This is the same procedure except that it's performed on an earlier block, therefore I expect it to have the same costs.